Reenable cni-calico-deep integration test

alpeb opened the pull request:
Fixes #11567

The trick is to run the test under k8s `v1.27.6-k3s1` as the following versions break the calico CNI plugin in k3s (see k3d-io/k3d#1375).

Also removed the `continue-on-error: true` directive in the integration workflow because it was hiding this problem.

olix0r commented:
We have continue-on-error set on other workflows. Should we? Why should this job be handled differently?

How was the error being hidden such that changing continue-on-error will prevent that problem?

olix0r commented:
I've made a separate change for the continue-on-error behavior: https://github.com/linkerd/linkerd2/pull/11630

olix0r commented on bin/_k3d-config.yml:
```diff
@@ -0,0 +1,3 @@
+apiVersion: k3d.io/v1alpha4
+kind: Simple
+image: rancher/k3s:v1.27.6-k3s1
```
As a favor to our future selves, can we rename this file so that it clearly indicates what we're using it for? This isn't a general purpose k3d config.

What about something like `test/integration/deep/calico-k3d.yml`?

And then in the file we should include a comment that references the k3x issue we're tracking.

alpeb replied:
Sounds good. The k3d issue was already referred to on the calling side, but I can add it to the config file as well.

alpeb commented:
In my latest commit I've moved the k3d config file and refactored a bit `bin/tests` and `bin/_test-helpers.sh`, removing the duped declaration of `bindir` and renaming `test_directory` to `testdir`.
